From f2d21ca42a23a14d9acccdc1a1415ef2c8cadfa7 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Thu, 5 Aug 2021 12:49:46 -0700
Subject: [PATCH] fuse: Initialize err before use in fuse_fill_super()

This is already fixed in the latest fuse/for-next tree so this is not
going to be submitted upstream.

Fixes: ff12b2314569 ("fuse: move fget() to fuse_get_tree()")
Link: https://lore.kernel.org/r/202108050030.cxg6NKHX-lkp@intel.com/
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 fs/fuse/inode.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/fuse/inode.c b/fs/fuse/inode.c
index 0580a8319593..1177bba769d1 100644
--- a/fs/fuse/inode.c
+++ b/fs/fuse/inode.c
@@ -1520,6 +1520,7 @@ static int fuse_fill_super(struct super_block *sb, struct fs_context *fsc)
 	 * Require mount to happen from the same user namespace which
 	 * opened /dev/fuse to prevent potential attacks.
 	 */
+	err = -EINVAL;
 	if ((ctx->file->f_op != &fuse_dev_operations) ||
 	    (ctx->file->f_cred->user_ns != sb->s_user_ns))
 		goto err;
-- 
2.33.0.rc0

